lib/chef/provider/windows_path.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Author:: Nimisha Sharad (<nimisha.sharad@msystechnologies.com>)
+# Copyright:: Copyright 2008-2017, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/mixin/windows_env_helper" if Chef::Platform.windows?
+require "chef/mixin/wide_string"
+require "chef/exceptions"
+
+class Chef
+  class Provider
+    class WindowsPath < Chef::Provider
+
+      include Chef::Mixin::WindowsEnvHelper if Chef::Platform.windows?
+
+      def load_current_resource
+        @current_resource = Chef::Resource::WindowsPath.new(new_resource.name)
+        @current_resource.path(new_resource.path)
+        @current_resource
+      end
+
+      action :add do
+        converge_by "Adding #{new_resource.path} to path environment variable" do
+          declare_resource(:env, "path") do
+            action :modify
+            delim ::File::PATH_SEPARATOR
+            value new_resource.path.tr("/", '\\')
+            notifies :run, "ruby_block[fix ruby ENV['PATH']]", :immediately
+          end
+
+          # The windows Env provider does not correctly expand variables in
+          # the PATH environment variable. Ruby expects these to be expanded.
+          #
+          declare_resource(:ruby_block, "fix ruby ENV['PATH']") do
+            block do
+              ENV['PATH'] = expand_path(ENV['PATH'])
+            end
+            action :nothing
```

Comment:
Noticed that the environment variable string isn't getting expanded even after calling `expand_path` method. The reason is that when this block executes, `ENV['PATH']` still doesn't have user provided path appended. Working on it.